jest/setupJestMock.js

```diff
@@ -13,9 +13,48 @@ global.SimVar.GetSimVarValue = jest.fn((name, _, __) => {
         return 0;
     }
 });
-global.SimVar.SetSimVarValue = jest.fn((name, _, value, __) => {
+
+global.SimVar.SetSimVarValue = jest.fn((name, _, value) => {
     return new Promise((resolve, _) => {
         values[name] = value;
         resolve();
     });
 });
+
+let listeners = {};
+
+global.beforeEach(() => {
+    listeners = {};
+});
+
+/** @type {Partial<ViewListener.ViewListener>} */
+const MockViewListener = {
+    triggerToAllSubscribers: jest.fn((event, ...args) => {
+        const subs = listeners[event];
+
+        const jsonArgs = args.map((it) => JSON.stringify(it)).map((it) => JSON.parse(it));
```

Comment:
**thought**: You copy the arguments. If your goal is to protect the caller of `triggerToAllSubscribers` from suddenly having its own objects changed under him, then this works. If your goal is for every individual subscriber to have their own copy of objects, then this will need to move within the loop.

Reply:
The goal here is to reproduce what MSFS does (value -> json -> value), and lose the prototypes and object identities.